js/util/url-query.ts

```diff
@@ -0,0 +1,42 @@
+import queryString from "query-string";
+
+/**
+ * Simplifies query-string library by only returning `string | undefined`, instead
+ * of `string | string[] | null | undefined`.
+ * @param prop
+ */
+export const queryValue = (prop: string): string | undefined => {
+  const query = queryString.parse(window.location.search);
+  const val = query[prop];
+  if (!val) return;
+  if (Array.isArray(val)) {
+    throw `May only have one query parameter for ${prop}. Found: ${val}`;
+  }
+  return val;
+};
```

Comment:
Note that we have urlParsing implemented in this repo, in `util/misc.ts`. It'd be good to reuse it if possible and/or add there more functions if they're useful.